frontend/dockerfile/dockerfile_runsecurity_test.go

```diff
@@ -0,0 +1,62 @@
+// +build dfrunsecurity
+
+package dockerfile
+
+import (
+	"context"
+	"os"
+	"testing"
+
+	"github.com/containerd/continuity/fs/fstest"
+	"github.com/moby/buildkit/client"
+	"github.com/moby/buildkit/frontend/dockerfile/builder"
+	"github.com/moby/buildkit/util/entitlements"
+	"github.com/moby/buildkit/util/testutil/integration"
+	"github.com/stretchr/testify/require"
+)
+
+var runSecurityTests = []integration.Test{
+	testRunSecurityInsecure,
+}
+
+func init() {
+	securityTests = append(securityTests, runSecurityTests...)
+
+}
+
+func testRunSecurityInsecure(t *testing.T, sb integration.Sandbox) {
+	f := getFrontend(t, sb)
+
+	dockerfile := []byte(`
+FROM busybox
+RUN --security=insecure [ "$(cat /proc/self/status | grep CapEff)" == "CapEff:	0000003fffffffff" ]
```

Comment:
Make sure there is a test that checks that this is not set when `--security=insecure` is not specified.

Reply:
added another test for that, when entitlements are allowed, solver fails if insecure is not enabled server side still

Reply:
nit: maybe `CapBnd` would be slightly better to check instead

Reply:
thanks, @tonistiigi updated